src/test/test-str-slice.c

```diff
@@ -108,6 +108,80 @@ test_str_slice_str_eq(void)
 #undef TEST_NOT_EQUAL
 }
 
+DEFINE_TEST(test_str_slice_remove_leading_whitespace);
+
+static void
+test_str_slice_remove_leading_whitespace(void)
+{
+    unsigned int i;
+
+#define TEST_EQUAL(X) { SOL_STR_SLICE_LITERAL(X), true }
+#define TEST_NOT_EQUAL(X) { SOL_STR_SLICE_LITERAL(X), false }
+
+    static const struct {
+        struct sol_str_slice input;
+        bool equal;
+    } table[] = {
+        TEST_NOT_EQUAL(" with one leading whitespace"),
+        TEST_NOT_EQUAL("  with two leading whitespace"),
+        TEST_NOT_EQUAL(" "),
+        TEST_NOT_EQUAL("\twith one leading whitespace"),
+        TEST_NOT_EQUAL("\t\twith two leading whitespace"),
+        TEST_NOT_EQUAL("\t"),
+        TEST_NOT_EQUAL("\nwith one leading whitespace"),
+        TEST_NOT_EQUAL("\n\nwith two leading whitespace"),
+        TEST_NOT_EQUAL("\n"),
+        TEST_EQUAL(""),
+        TEST_EQUAL("without leading whitespace"),
```

Comment:
Please add the test I've asked for in the previous PR. It'll ensure that a buffer underflow won't happen when executed with address sanitizer or Valgrind (which the bots do by default):

     const char test_underflow_buf[] = "         some random string";
     const char test_underflow_no_leading_space[] = "some random string";
     sol_str_slice test_underflow = sol_str_slice_from_str(test_underflow_buf + 3);

     ASSERT(sol_str_slice_eq(test_underflow, test_underflow_no_leading_space));